misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -432,10 +432,7 @@ internal class RealTransacter private constructor(
       check(!readOnly) { "Saving isn't permitted in a read only session." }
       return when (entity) {
         is DbChild<*, *> -> (hibernateSession.save(entity) as Gid<*, *>).id
-        is DbRoot<*> -> hibernateSession.save(entity)
-        is DbUnsharded<*> -> hibernateSession.save(entity)
-        else -> throw IllegalArgumentException(
-            "You need to sub-class one of [DbChild, DbRoot, DbUnsharded]")
+        else -> hibernateSession.save(entity)
```

Comment:
Does the transacter know if its in a sharded environment? If it does, then you could leave the old logic if there are shards or keyspaces, and allow `DbEntity` otherwise.

Reply:
I don't think it'll make a difference here, hibernate handles them the same way.